network/network.go

```diff
@@ -1,3 +1,8 @@
+// Package network provides core networking abstractions for libp2p.
+//
+// The network package provides the high-level Network interface for interacting
+// with other libp2p peers, which is the primary public API for initiating and
+// accepting connections to remote peers.
```

Comment:
The Network is generally a confusing component for users. We'll need to expand on this description. Possible content: The Network is the component that interacts with the underlying physical network stack in order to establish outbound connections and accept inbound connections to/from peers.